lib/string_reverse.rb

```diff
@@ -1,4 +1,18 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  # checks to see if there is a string to reverse
+  return nil if my_string == nil || my_string.length == 0
+
+  string = 0 # first character of string
+  reverse_string = my_string.length - 1 # last character of string

```

Comment:
I'd recommend naming your variables something that makes it easier to know what they stand for. e/g/ instead of calling the variable which is the index of the first character of the string `string`, you could call it `forward_index` or simply `i` and instead of `reverse_string`, call the index of the last character of string, something like `backwards_index` or simply `j`.